src/go/rpk/pkg/cli/redpanda/admin/brokers/list.go

```diff
@@ -26,27 +50,88 @@ func newListCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 			bs, err := cl.Brokers(cmd.Context())
 			out.MaybeDie(err, "unable to request brokers: %v", err)
 
-			headers := []string{"Node-ID", "Num-Cores", "Membership-Status"}
+			headers := []string{"ID", "Host", "Port", "Rack", "Cores", "Membership", "Is-Alive", "Version"}
 
 			args := func(b *rpadmin.Broker) []interface{} {
-				ret := []interface{}{b.NodeID, b.NumCores, b.MembershipStatus}
+				ret := []interface{}{b.NodeID, b.InternalRPCAddress, b.InternalRPCPort, checkEmpty(b.Rack), b.NumCores, b.MembershipStatus, *b.IsAlive, extractVersion(b.Version)}
 				return ret
 			}
-			for _, b := range bs {
-				if b.IsAlive != nil {
-					headers = append(headers, "Is-Alive", "Broker-Version")
-					orig := args
-					args = func(b *rpadmin.Broker) []interface{} {
-						return append(orig(b), *b.IsAlive, b.Version)
-					}
-					break
+
+			idUUIDMapping, err := cl.GetBrokerUuids(cmd.Context())
+			if err != nil {
+				fmt.Printf("unable to retrieve node UUIDs: %v", err)
+			}
+			if idUUIDMapping != nil {
+				headers = append(headers, "UUID")
+				org := args
+				args = func(b *rpadmin.Broker) []interface{} {
+					return append(org(b), mapUUID(b.NodeID, idUUIDMapping))
 				}
 			}
+
 			tw := out.NewTable(headers...)
 			defer tw.Flush()
 			for _, b := range bs {
 				tw.Print(args(&b)...)
 			}
+
+			if all && idUUIDMapping != nil {
+				decomNodes := extractDecomNodes(bs, idUUIDMapping)
+				for _, b := range decomNodes {
+					tw.Print(b.NodeID, "-", "-", "-", "-", "-", "-", "-", b.UUID)
+				}
+			}
 		},
 	}
+	cmd.Flags().BoolVarP(&all, "all", "a", false, "If true, include decommissioned brokers")
+	return cmd
+}
+
+// mapUUID returns a UUID from "mapping" which node ID maps to "nodeID".
+func mapUUID(nodeID int, mapping []rpadmin.BrokerUuids) string {
+	var UUIDs []string
+	for _, node := range mapping {
+		if nodeID == node.NodeID {
+			UUIDs = append(UUIDs, node.UUID)
+		}
+	}
+	if len(UUIDs) == 0 {
+		return "UUID Not Found"
```

Comment:
This means we will have something like

```
ID    HOST       PORT   RACK  CORES  MEMBERSHIP  IS-ALIVE  VERSION  UUID
0     localhost  33146        2      active      true      v24.2.3  UUID Not Found
```

Wonder if we should use `-` as in RACK, or decommissioned nodes. In which cases will we have empty node UUIDs?

Reply:
Rack is optional whereas missing UUID is an unexpected scenario, which indicates a core issue. Thus I wanted to distinguish each other. Should I make it more explicit when UUID is not found?

Reply:
It should be something like `FATAL: UUID Not Found`?

Reply:
After some thought, I’m leaning towards using a hyphen when a UUID is not found to make it more awk-friendly. I believe it is enough to recognize the issue with hyphen.

Reply:
Great, thanks!